doc/reset_factory/ResetFactoryHLD.md

```diff
@@ -0,0 +1,202 @@
+		* 1.5.2. [Configuration and Management Requirements](#ConfigurationManagementRequirements) 
+* 2. [Design](#Design)
+	* 2.1. [High-Level Design](#High-LevelDesign)
+* 3. [Functionality](#Functionality)
+		* 3.1. [Reset-Factory](#Reset-Factory)
+		* 3.2. [Config-setup Factory](#Config-setup-Factory) 
+* 4. [CLI](#CLI)
+* 5. [Restrictions/Limitations](#RestrictionsLimitations)
+* 6. [Test Plan](#TestPlan)
+		* 6.1. [Unit Test cases](#UnitTestcases)
+
+
+
+###  1.1. Revision
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 01/2023 | Mohammed Zayadna | Phase 1 Design     |
+
+###  1.2. <a name='Scope'></a>Scope
+
+This HLD document described the requirements, architecture details of Reset factory feature in switches Sonic OS based.
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations
+N/A
+
+
+###  1.4. <a name='Overview'></a>Overview
+
+A factory reset is a process that restores the switch to its original manufacturer settings. This can be useful if the device's configuration has become corrupt or if the user wishes to erase all personalized configurations.
+
+
+###  1.5. <a name='Requirements'></a>Requirements
+####  1.5.1. <a name='FunctionalRequirements'></a>Functional requirements
+1. It should support resetting configuration to factory default settings.
+
+2. It should clear logs and tech-support files, users history files and home directories.
+
+3. It should reboot the system to load and apply the new configuration.
+
+4. It should stop sonic.target to ensure that files are not deleted while other daemons are utilizing them.
+
+5. it should allow users/application to create their own factory default configurations or
+extend the reset factory functionality.
+
+
+####  1.5.2. <a name='ConfigurationManagementRequirements'></a>Configuration and Management Requirements
+We will create a command line tool /usr/bin/reset-factory which should provide following functionality:
+
+1. default: reset configurations to factory default, clear all system logs and files.
+
+2. keep-basic: preserves basic configurations only after boot. Logs and files will be deleted.
+
+3. keep-all-config: preserves all configurations after boot. Logs and files will be deleted.
+
+4. only-config: reset configurations to factory default. Logs and files will be preserved.
+
+    
+##  2 <a name='Design'></a>Design
+	
+###  2.1 <a name='High-LevelDesign'></a>High-Level Design
+![diagram](diagram.png)
+
+(flow description in the chapter below)
+
+"reset-factory" script will create new config_db.json if needed using "config-setup factory" command.<br/>
+In addition, it will clear system logs and files.<br/>
+
+Steps:
+
+1. Stops sonic.target: 
+
+systemctl stop sonic.target
+
+2. If not "keep-all-config":
+
+config-setup factory <None/keep-basic/only-config>
+
+3. If not ("only-config" or “keep-basic”):
+
+* Delete bash , vim and python history files under "/home".
```

Comment:
Will there be a warning to the user about these?

Reply:
Sure